sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysisProvider.java

```diff
@@ -52,9 +53,10 @@
                                                      String workingStateId,
                                                      SensitivityFactorsProvider factorsProvider,
```

Comment:
SensitivityFactorsProvider has to be removed from the new API. Primary API so the one that should be implemented in the `SensitivityAnalysisProvider` has to rely on `SensivitiyFactorReader` and `SensitivityValueWriter`, nothing else.
In  `SensitivityAnalysis` we should have a run method also based on `SensivitiyFactorReader` and `SensitivityValueWriter` and another one for simple purpose based in a `List<SensivityFactor>` + `SensivityAnlysisResult` in input using the bridge between reader/writer and objects `SensivityFactor`/`SensivityValue`.

Reply:
@geofjamg This should be covered by my last commit.